Begin cleanup/refactor

jtsternberg opened the pull request:
beginning some cleanup/refactor, etc. From my testing, didn't break any existing functionality, but would prefer further testing in layouts/areas.

Also, we need to continue cleaning up/refactoring. The areas/layouts (and even new fields registration bit) are a bit tacked on, and need to be better architected so that they don't conflict with other parts.

jtsternberg committed:
Add wds_page_builder_load_template_part filter to for conditionally loading template parts

jtsternberg committed:
Begin cleanup/refactor

jazzsequence commented:
Agreed. Areas are _definitely_ tacked on. Layouts a bit less so, but could be handled better.
Somewhat related: https://github.com/WebDevStudios/WDS-Simple-Page-Builder/issues/27

JayWood committed:
Fix $$ declaration

blobaugh committed:
Added type check for array

The type expected from get_all() was expected to be an array but the method
may not always return an array. This conditional check simply ensures valid
data is available to work with
